File: algorithms/easy/minimum_pair_removal_to_sort_array_i.rs
fn is_sorted(v: &Vec<i32>) -> bool {
    for i in 1..v.len() {
        if v[i - 1] > v[i] {
            return false;
        }
    }
    true
}

impl Solution {
    pub fn minimum_pair_removal(mut nums: Vec<i32>) -> i32 {
        let mut ct = 0;
        while !is_sorted(&nums) {
            let mut sum = i32::MAX;
            let mut j = 0;
            for i in 1..nums.len() {
                if nums[i] + nums[i - 1] < sum {
                    sum = nums[i] + nums[i - 1];
                    j = i;
                }
            }
            nums[j - 1] += nums[j];
            nums.remove(j);
            ct += 1;
        }
        ct
    }
}

File: algorithms/easy/minimum_pair_removal_to_sort_array_i.kt
class Solution {
  fun isSorted(nums: MutableList<Int>): Boolean {
    for (i in 1..<nums.size) {
      if (nums[i] < nums[i - 1]) {
        return false
      }
    }
    return true
  }

  fun minimumPairRemoval(nums: IntArray): Int {
    val arr = nums.toMutableList()
    var ct = 0

    while (!isSorted(arr)) {
      var minSum = Int.MAX_VALUE
      var minIndex = 0
      for (i in 1..<arr.size) {
        val sum = arr[i - 1] + arr[i]
        if (sum < minSum) {
          minSum = sum
          minIndex = i - 1
        }
      }
      arr[minIndex] = minSum
      arr.removeAt(minIndex + 1)
      ct += 1
    }

    return ct
  }
}
